src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+		    ddl_audit_handle->commit_count, ddl_audit_handle->file_line_number);
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->pid, ddl_audit_handle->user_name, result, ddl_audit_handle->msg,
+			 ddl_audit_handle->schema_file);
+    }
+  else if (ddl_audit_handle->app_name == APP_NAME_CSQL)
+    {
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	  ddl_audit_handle->elapsed_time[0] = '\0';
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->pid,
+			 ddl_audit_handle->user_name,
+			 result, ddl_audit_handle->elapsed_time, ddl_audit_handle->msg, ddl_audit_handle->sql_text);
+    }
+  else
+    {
+      if (ddl_audit_handle->auto_commit_mode == FALSE)
+	{
+	  gettimeofday (&log_time, NULL);
+	  cub_get_time_string (ddl_audit_handle->execute_start_time, &log_time);
+	}
+
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      if (ddl_audit_handle->execute_type & DDL_LOG_RUN_EXECUTE_BATCH_FUNC)
+	{
+	  snprintf (ddl_audit_handle->elapsed_time, sizeof (ddl_audit_handle->elapsed_time), "elapsed time 0.000");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %s|%s|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->ip_addr,
+			 ddl_audit_handle->user_name,
+			 result, ddl_audit_handle->elapsed_time, ddl_audit_handle->msg, ddl_audit_handle->sql_text);
+
+    }
+
+  if (retval >= DDL_LOG_BUFFER_SIZE)
+    {
+      msg[DDL_LOG_BUFFER_SIZE - 2] = '\n';
+      msg[DDL_LOG_BUFFER_SIZE - 1] = '\0';
+      retval = DDL_LOG_BUFFER_SIZE;
+    }
+  return retval;
+}
+
+static void
+cub_ddl_log_backup (const char *path)
+{
+  char backup_file[PATH_MAX] = { 0 };
+#if !defined(WINDOWS)
+  sigset_t new_mask, old_mask;
+#endif /* !WINDOWS */
+
+  snprintf (backup_file, sizeof (backup_file), "%s.bak", path);
+
+#if !defined(WINDOWS)
+  sigfillset (&new_mask);
+  sigdelset (&new_mask, SIGINT);
+  sigdelset (&new_mask, SIGQUIT);
+  sigdelset (&new_mask, SIGTERM);
+  sigdelset (&new_mask, SIGHUP);
+  sigdelset (&new_mask, SIGABRT);
+  if (sigprocmask (SIG_SETMASK, &new_mask, &old_mask) == 0)
+    {
+      unlink (backup_file);
+      rename (path, backup_file);
+      sigprocmask (SIG_SETMASK, &old_mask, NULL);
+    }
+#else /* !WINDOWS */
+  unlink (backup_file);
+  rename (path, backup_file);
+#endif
+}
+
+static FILE *
+cub_fopen_and_lock (const char *path, const char *mode)
+{
+#define MAX_RETRY_COUNT 100
+  int retry_count = 0;
+  FILE *ddl_log_fd = NULL;
```

Comment:
What about
`FILE *ddl_log_fp = NULL;`
It is file pointer, not descriptor.

Reply:
Thank you, i will change code as your recommend.